app/src/main/java/com/example/cleanarchitecture/fragment/CharacterInformationFragment.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.cleanarchitecture.fragment
+
+import android.os.Bundle
+import android.text.method.ScrollingMovementMethod
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.bumptech.glide.Glide
+import com.example.cleanarchitecture.R
+import com.example.cleanarchitecture.adapter.ComicAdapter
+import com.example.cleanarchitecture.databinding.FragmentCharacterInformationBinding
+import com.example.cleanarchitecture.utils.Event
+import com.example.cleanarchitecture.viewmodel.CharacterInformationViewModel
+import com.example.domain.entity.CharacterData
+import java.io.Serializable
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class CharacterInformationFragment : DialogFragment() {
+
+    private lateinit var binding: FragmentCharacterInformationBinding
+    private val viewModel by viewModel<CharacterInformationViewModel>()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = FragmentCharacterInformationBinding.inflate(inflater, container, false)
+        viewModel.liveDataState.observe(this, { updateUI(it) })
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        arguments?.let {
+            viewModel.showCharacterInformation(it.getSerializable(CHARACTER_DATA_KEY) as CharacterData)
+        }
+    }
+
+    private fun updateUI(comicData: Event<CharacterData>) {
+        with(binding) {
+            imageButtonFragmentClose.setOnClickListener { dismiss() }
+            comicData.getContentIfNotHandled()?.let {
+                textViewFragmentTitle.text = context?.getString(R.string.text_view_title_message_fragment, it.name)
+                if (it.description.isNotEmpty()) {
+                    textViewFragmentDescription.text = context?.getString(R.string.text_view_description_message_fragment, it.description)
+                    textViewFragmentDescription.movementMethod = ScrollingMovementMethod()
+                } else {
+                    textViewFragmentDescription.text = context?.getString(R.string.text_view_default_description_message_fragment)
+                }
+                Glide.with(root)
+                    .load("${it.thumbnail.path}.${it.thumbnail.extension}")
+                    .into(imageViewFragmentIcon)
+                recyclerViewFragmentComics.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+                recyclerViewFragmentComics.adapter = ComicAdapter(it.comics.items)
+            }
+        }
```

Comment:
you need to do this into the EVENT , not outside

Reply:
done